R/leaflet.R

```diff
@@ -19,15 +19,24 @@
 #' @param width the width of the map
 #' @param height the height of the map
 #' @param padding the padding of the map
+#' @param mapOptions the map options
 #' @return A HTML widget object, on which we can add graphics layers using
 #'   \code{\%>\%} (see examples).
 #' @example inst/examples/leaflet.R
 #' @export
-leaflet = function(data = NULL, width = NULL, height = NULL, padding = 0) {
+leaflet = function(data = NULL, width = NULL, height = NULL,
+                   padding = 0, mapOptions = list()) {
```

Comment:
Agreed on renaming mapOptions argument to options.

 But as for introducing a mapOptions() function, there's already a mapOptions function, which behaves differently than other options() function.  tileOptions(), markerOptions() etc. all give back a list . But mapOptions as it's implemented now returns a map. Here's the current implementation

```r
function (map, zoomToLimits = c("always", "first", "never"))
{
    if (is.null(map$x$options))
        map$x$options <- list()
    zoomToLimits <- match.arg(zoomToLimits)
    map$x$options$zoomToLimits <- zoomToLimits
    map
}
```

If I make it similar to other Options() functions it breaks compatibility.